examples/kubernetes/README.rst

```diff
@@ -0,0 +1,58 @@
+Kubernetes Deployment
+=====================
+
+This directory contains all Cilium deployment files that can be used in
+Kubernetes.
+
+Each directory represents a Kubernetes version, from :code:`1.7` to :code:`1.11`,
+and inside each version there is a list of files to deploy Cilium.
+
+The structure directory will be :code:`${k8s_major_version}.${k8s_minor_version}/*.yaml`.
+
+To generate those files simply run :code:`make`, which will pick up the Cilium
+version from the :code:`VERSION` file at the root of the repo, or if you want to
+specify the Cilium version yourself use :code:`make CILIUM_VERSION=X.Y.Z`.
+
+If you want to clean up a specific version, run :code:`make CILIUM_VERSION=X.Y.Z`
+which will delete all generated files for a specific Cilium version
+
+Templates
+---------
+
+There are templates for each component to be installed in Kubernetes inside
+the directory :code:`templates`. The components ending with :code:`.sed` will be
+automatically generated based on the template itself and the specific
+:code:`transforms2sed.sed` inside each directory for each Kubernetes version.
+
+Files
+-----
+
+Inside each :code:`${k8s_major_version}.${k8s_minor_version}` directory there
+are 5 files:
+
+- :code:`cilium-cm.yaml` - The :code:`ConfigMap` and options with some default
+  values the user should change
+
+- :code:`cilium-ds.yaml` - The :code:`DaemonSet` to deploy Cilium in the
+  Kubernetes cluster, some advanced options can be changed here.
+
+- :code:`cilium-rbac.yaml` - The Cilium's RBAC for the Kubernetes cluster.
+
+- :code:`cilium-sa.yaml` - The Cilium's Kubernetes :code:`ServiceAccount`.
+
+- :code:`cilium.yaml` - All previous files concatenated into a single file,
+  useful to deploy Cilium in a minikube environment with a "single line" command.
```

Comment:
I guess we need this because we can't run `kubectl -f` on a url directory. Do we want to support running it on an on-disk copy? If so, we'd need to move this file out of the k8s version specific dir. Perhaps:
```
/1.8/cilium-{cm,ds,rbac,sa}.yaml
/cilium-1.8.yaml
```

It's probably not worth it, however, since it makes all our generators more complex.

Reply:
oh, my reasoning was that you can then do the `-f path/to/dir` if you have a local directory, but can still use `-f https://githubrawwhatever.com/cilium/cilium-1.8.yaml` for the tutorial examples. This cilium-1.8.yaml file is at the same level as the 1.8 directory.